tensorflow/python/eager/def_function.py

```diff
@@ -773,7 +773,41 @@ def _decorate(self, decorator):
     self._function_spec = function_lib.FunctionSpec.from_function_and_signature(
         self._python_function, self.input_signature)
 
-  def _get_tracing_count(self):
+  def experimental_get_tracing_count(self):
+    """Returns the number of times the function has been traced.
+
+    A function may be traced more than once.
+
+    For example, as of now a function that was traced using the
+    tf.function decorator would be retraced if called with arguments
+    of different types wrt any previous call of said function.
+    
+    For example if this function is defined:
+
+    ```@tf.function
```

Comment:
We have a mechanism that ensures these snippets are tested. It uses [doctest](https://docs.python.org/3/library/doctest.html). Let's write the code snippets using that syntax as well.

Here's an example:

https://github.com/tensorflow/tensorflow/blob/70e52bd49af344df268c25077a00b0b758982f16/tensorflow/python/eager/def_function.py#L1315

Also, feel free to point to the tf.function documentation to make sure the information is not duplicated.

Reply:
Thanks again! I will push the new changes as soon as I can. The topic is discussed in this guide https://www.tensorflow.org/guide/function. I suppose it is fine to point to this guide instead?

Reply:
Yep that sounds good.